test/lib/BedrockTester.cpp

```diff
@@ -622,6 +622,24 @@ bool BedrockTester::waitForStatusTerm(const string& term, const string& testValu
     return false;
 }
 
+bool BedrockTester::waitForLeadingFollowing(uint64_t timeoutUS) {
+    uint64_t start = STimeNow();
+    while (STimeNow() < start + timeoutUS) {
+        try {
+            string result = SParseJSONObject(BedrockTester::executeWaitVerifyContent(SData("Status"), "200", true))["state"];
+
+            // if the value matches, return, otherwise wait
+            if (result == "LEADING" || result == "FOLLOWING") {
+                return true;
+            }
+        } catch (...) {
+            // Doesn't do anything, we'll fall through to the sleep and try again.
+        }
+        usleep(100'000);
+    }
+    return false;
+}
+
```

Comment:
This is added just so that we wan wait for either LEADING or FOLLOWING instead of just one or the other.